File: api/PaymentGatewayDemo.Domain/Models/Billing.cs
using System.ComponentModel.DataAnnotations;

namespace PaymentGatewayDemo.Domain.Models;

public class Billing
{
    [Key] public string TransactionId { get; set; }

    public Guid ProductKeyId { get; set; }
    public Product? Product { get; set; } = null;

    public string Title { get; set; }
    public decimal Price { get; set; }
    public string Status { get; set; }

    public string UserId { get; set; }
    public User User { get; set; }

    public DateTimeOffset CreationDate { get; set; }
    public DateTimeOffset? RealizationDate { get; set; }
}

File: web/src/types/billing/billing.ts
export type Billing = {
  transactionId: string;
  productKeyId: string;
  title: string;
  friendlyTitle: string;
  price: number;
  status: string;
  userId: string;
  creationDate: string;
  realizationDate?: string | null;
};

export type BillingDictionary = Record<string, Billing[]>;
